core/src/main/java/org/mule/api/metadata/MetadataAware.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.metadata;
+
+import org.mule.api.metadata.descriptor.OperationMetadataDescriptor;
+import org.mule.extension.api.annotation.param.metadata.Content;
+import org.mule.extension.api.metadata.MetadataResolver;
+import org.mule.extension.api.metadata.MetadataResolvingException;
+import org.mule.metadata.api.model.MetadataType;
+
+import java.util.List;
+
+/**
+ * Provides access to the Processor's Metadata resolving methods.
+ */
+public interface MetadataAware
```

Comment:
is it me or there's a lot of overlap between this interface and the MetadataManager?